Tests fail after quarkus:dev run

When running mvn package (no clean) after quarkus:dev, 1. tests fail 2. with a weird inlined exception in the console.

```
# build Quarkus master
./mvn clean install -DskipTests

# create new project
cd /tmp
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:create -DprojectGroupId=org.acme -DprojectArtifactId=test -DclassName="org.acme.GreetingResource" -Dpath="/greeting"   -DplatformArtifactId=quarkus-bom -DplatformVersion=999-SNAPSHOT

# run quarkus:dev
cd test
./mvn clean compile quarkus:dev

# CTRL+C
# start package
./mvnw package
^[[A^[[A ~  Dow  nob  banner-test  mvn package
[INFO] Scanning for projects...
[INFO]
[INFO] ------------------------< org.acme:banner-test >------------------------
[INFO] Building banner-test 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO]
[INFO] --- maven-resources-plugin:2.6:resources (default-resources) @ banner-test ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] Copying 2 resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:compile (default-compile) @ banner-test ---
[INFO] Nothing to compile - all classes are up to date
[INFO]
[INFO] --- maven-resources-plugin:2.6:testResources (default-testResources) @ banner-test ---
[INFO] Using 'UTF-8' encoding to copy filtered resources.
[INFO] skip non existing resourceDirectory /Users/emmanuel/Downloads/nobackup/banner-test/src/test/resources
[INFO]
[INFO] --- maven-compiler-plugin:3.8.1:testCompile (default-testCompile) @ banner-test ---
[INFO] Changes detected - recompiling the module!
[INFO] Compiling 2 source files to /Users/emmanuel/Downloads/nobackup/banner-test/target/test-classes
[INFO]
[INFO] --- maven-surefire-plugin:2.22.1:test (default-test) @ banner-test ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running org.acme.GreetingResourceTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.28 s <<< FAILURE! - in org.acme.GreetingResourceTest
[ERROR] testHelloEndpoint  Time elapsed: 0.011 s  <<< ERROR!
java.lang.RuntimeException: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
Caused by: java.lang.RuntimeException: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype
Caused by: java.util.ServiceConfigurationError: org.eclipse.microprofile.config.spi.ConfigSourceProvider: io.quarkus.test.common.http.TestHTTPConfigSourceProvider not a subtype

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   GreetingResourceTest.testHelloEndpoint » Runtime java.lang.RuntimeException: j...
[INFO]
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.166 s
[INFO] Finished at: 2020-03-04T21:19:05+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.1:test (default-test) on project banner-test: There are test failures.
[ERROR]
[ERROR] Please refer to /Users/emmanuel/Downloads/nobackup/banner-test/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
[ERROR]
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR]
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

If `./mvnw clean package` is run, then no failure occurs.

--- a/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java
+++ b/independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/BootstrapAppModelFactory.java
@@ -46,7 +46,7 @@ public class BootstrapAppModelFactory {
 
     private static final String QUARKUS = "quarkus";
     private static final String BOOTSTRAP = "bootstrap";
-    private static final String DEPLOYMENT_CP = "deployment.cp";
+    private static final String APP_MODEL_DAT = "app-model.dat";
 
     public static final String CREATOR_APP_GROUP_ID = "creator.app.groupId";
     public static final String CREATOR_APP_ARTIFACT_ID = "creator.app.artifactId";
@@ -441,8 +441,10 @@ public class BootstrapAppModelFactory {
         }
     }
 
-    private static Path resolveCachedCpPath(LocalProject project) {
-        return project.getOutputDir().resolve(QUARKUS).resolve(BOOTSTRAP).resolve(DEPLOYMENT_CP);
+    private Path resolveCachedCpPath(LocalProject project) {
+        final String filePrefix = test ? "test-" : (devMode ? "dev-" : null);
+        return project.getOutputDir().resolve(QUARKUS).resolve(BOOTSTRAP)
+                .resolve(filePrefix == null ? APP_MODEL_DAT : filePrefix + APP_MODEL_DAT);
     }
 
     private static void debug(String msg, Object... args) {

